### PR TITLE
fix: Fix the transport bazel arg (it's not a file)

### PR DESCRIPTION
**Pull request by jskeet**
At the same time, this moves to using opt_args and opt_file_args to
generate cleaner protoc invocations.

### CODE REVIEW

**Comment by jskeet** [comment]
Unfortunately the 1.45 release was broken - this one definitely *does* work. (I've tested it against the failing googleapis commit.)

File: rules_csharp_gapic/csharp_gapic.bzl [hunk 1]
```diff
@@ -47,22 +47,24 @@ def csharp_gapic_library(
...
+        opt_file_args[grpc_service_config] = "grpc-service-config"
     if common_resources_config:
-        plugin_file_args[common_resources_config] = "common-resources-config"
+        opt_file_args[common_resources_config] = "common-resources-config"
     if service_yaml:
-        plugin_file_args[service_yaml] = "service-config"
+        opt_file_args[service_yaml] = "service-config"
     if rest_numeric_enums:
-        plugin_file_args[rest_numeric_enums] = "rest-numeric-enums"
+        opt_file_args[rest_numeric_enums] = "rest-numeric-enums"
```
Comment (viacheslav-rostovtsev) [inline rules_csharp_gapic/csharp_gapic.bzl:59]: Probably also not a file

**Review by viacheslav-rostovtsev — APPROVED** [review]
LGTM (but check comments)

File: rules_csharp_gapic/csharp_gapic.bzl (hunk 1 above)
Comment (jskeet) [reply rules_csharp_gapic/csharp_gapic.bzl:59]: Thanks, will fix in the same way as the transport.

**Comment by jskeet** [comment]
I've now tested this with Compute, including with numeric enums.

**Pull request merged**